core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -100,7 +102,16 @@ private[spark] object JsonProtocol {
         executorMetricsUpdateToJson(metricsUpdate)
       case blockUpdate: SparkListenerBlockUpdated =>
         blockUpdateToJson(blockUpdate)
-      case _ => parse(mapper.writeValueAsString(event))
+      case _ =>
+        val outputStream = new PipedOutputStream()
+        val inputStream = new PipedInputStream(outputStream)
+        try {
+          mapper.writeValue(outputStream, event)
+          parse(inputStream)
+        } finally {
+          IOUtils.closeQuietly(outputStream)
```

Comment:
and .. another note for  `IOUtils.closeQuietly` saying that it's intentionally used in case that the close might be already attempted by Jackson's library if I understood correctly?